### DISCUSSION TITLE
Avoid clobbering/scrolling-up existing console content when using fzf search window

### DISCUSSION THREAD

**Discussion by jchia** [PatrickF1/fzf.fish#247] [Q&A]
When I press ctrl-r, ctrl-alt-f, etc to bring up an fzf search window to search for command-line history or files, the existing content in the console gets scrolled up to make space for the fzf search window. When the search window is closed, the terminal is not back to its original state; it is in a state where the original terminal content has been scrolled up by a number of lines equal to the height of the search window.

This recording demonstrates what I mean: https://asciinema.org/a/SkSAJ2LdhNElpnTz5Aoj17KlU

Is there a way to configure fzf.fish so that the terminal state is preserved across uses of the search window? It is disruptive to lose sight of my earlier terminal history (e.g. command-lines I ran earlier) so that I have to scroll up to see them.

fzf itself, if you run it from the command-line, preserves the terminal content.

I know I could hack/configure fzf.fish to use fzf-tmux instead of fzf to achieve this effect when I am already in a tmux. But what if I'm not in a tmux?

**Comment by PatrickF1** [comment]
Hi Joshua, I don't know how to stop it from erasing scrollback when the window opens up, but you could try making the window open down. Then I think your problem will be solved.

**Reply by jchia** [reply]
It's probably technically complicated to preserve the scrollback while having the current behavior when the fzf window is being displayed. For now, I've hacked _fzf_wrapper.fish to run `fzf --no-height` instead of `fzf`. I can't see anything from the original terminal content (including the last few lines from the terminal) while the fzf window is up; only the fzf window is displayed. However, the original terminal content is preserved upon exiting the fzf window. This is OK for me as I usually don't really need to see anything from the scrollback when using the fzf window.

**Reply by PatrickF1** [reply]
> For now, I've hacked _fzf_wrapper.fish to run fzf --no-height instead of fzf

Not sure what you mean by hack, but I just want to make sure you know there's the option of passing `--no-height` via `FZF_DEFAULT_OPTS` https://github.com/PatrickF1/fzf.fish#always-pass-some-options-to-fzf.

**Reply by jchia** [reply]
I mean I changed the 2nd last line in .config/fish/functions/_fzf_wrapper.fish, adding a ` --no-height` argument to the fzf command-line. It was an experiment. I am now using the original _fzf_wrapper.fish code but with a one-time `set -Ux FZF_DEFAULT_OPTS '--no-height'` to the same effect.
